### PR TITLE
mark xfail for float32

### LINKED ISSUE #2791: Robustify test_variational_inference.py::test_clear_cache
pymc3/tests/test_variational_inference.py::test_clear_cache failed (both locally and on Travis CI) 50% of the times, as `del inference` does not always delete the cache reliably.

### PR DESCRIPTION
**Pull request by junpenglao**
hotfix for #2791
cc @ferrine

### CODE REVIEW

File: pymc3/tests/test_variational_inference.py [hunk 1]
```diff
@@ -610,7 +610,7 @@ def test_remove_scan_op():
         assert 'theano.scan_module.scan_op.Scan' not in buff.getvalue()
         buff.close()
 
-
+@pytest.mark.xfail(condition=(theano.config.floatX == "float32"), reason="Fails on float32")
```
Comment (ferrine) [inline pymc3/tests/test_variational_inference.py:613]: god why?

**Comment by ferrine** [comment]
I think that's not a solution. We should try to clear cache reliably

**Comment by fonnesbeck** [comment]
Correct, it is not a solution. We should fix it, for sure, but if we can't prior to 3.3 this seems like a reasonable thing to do.

**Comment by junpenglao** [comment]
Said test seems to be passing fine lately. Closing this since it is not a real solution.

**Pull request closed without merging**